### PR TITLE
Generate #[allow(deprecated)] for methods for deprecated fields in the wrapper

### LINKED ISSUE #237: Fix warnings due to generated code
The generated protobufs cause lots of warnings with latest Prost (updated in https://github.com/tikv/client-rust/pull/232)

### PR DESCRIPTION
**Pull request by ekexium**
An attempt to fix tikv/client-rust#237.

I'm not sure if it's the best way to do it. PTAL @nrc

Test failed due to https://github.com/stepancheg/rust-protobuf/issues/551

### CODE REVIEW

File: src/wrapper.rs [hunk 1]
```diff
@@ -333,21 +337,22 @@ impl FieldKind {
...
-    fn methods(&self, ty: &Type, ident: &Ident) -> Option<FieldMethods> {
-        let mut result = FieldMethods::new(ty, ident);
+    fn methods(&self, ty: &Type, ident: &Ident, deprecated: bool) -> Option<FieldMethods> {
+        let mut result = FieldMethods::new(ty, ident, deprecated);
         match self {
             FieldKind::Optional(fk) => {
                 let unwrapped_type = unwrap_type(ty, "Option");
                 let unboxed_type = unwrap_type(&unwrapped_type, "Box");
-                let nested_methods = fk.methods(&unwrapped_type, ident).unwrap();
+                // FIXME: not sure whether to propagate here
```
Comment (nrc) [inline src/wrapper.rs]: Yes it should

**Review by nrc — APPROVED** [review]
LGTM, I would like to make it optional to generate deprecated fields, but if we do generate them, then we should do this, so this seems like a fine fix.

**Comment by nrc** [comment]
I'm going to put off doing a release until stepancheg/rust-protobuf#551 is fixed some how. If that doesn't happen in the next week or so, ping me and spin off a release.

**Comment by ekexium** [comment]
> I'm going to put off doing a release until [stepancheg/rust-protobuf#551](https://github.com/stepancheg/rust-protobuf/issues/551) is fixed some how. If that doesn't happen in the next week or so, ping me and spin off a release.

It should be fixed in rust-protobuf 2.22.1. Shall we release a new version?

**Comment by nrc** [comment]
I released 0.12.1

**Pull request merged**